### PR TITLE
Fix aroma beta 17 incompatibilities, fix debug logger, bump version

### PR DESCRIPTION
**Pull request by TraceEntertains**
Pretty self explanatory
Works fine on my wii u but will mark as a draft until confirmed working

### CODE REVIEW

**Comment by ashquarky** [comment]
Code looks good to me, though you've got mixed tabs/spaces. Might want to see if whatever editor your using has an [EditorConfig](https://github.com/PretendoNetwork/Inkay/blob/main/.editorconfig) plugin.

**Comment by TraceEntertains** [comment]
did a bunch of stuff but notifications still arent the way i wanted them to be, still a draft for now but i pushed all my latest changes so if anyone wants to take a look they can

**Comment by TraceEntertains** [comment]
> Code looks good to me, though you've got mixed tabs/spaces. Might want to see if whatever editor your using has an [EditorConfig](https://github.com/PretendoNetwork/Inkay/blob/main/.editorconfig) plugin.

i do actually have multiple ides installed but i just use notepad++ because it starts quickly and is just files rather than folder/project based

**Comment by TraceEntertains** [comment]
oh crap, the commit i put has debug defaulted to on (you can just comment out the "DEBUG := 1" line if you dont want debug but it will be commented before the pr is marked as ready for review)

**Comment by Maschell** [comment]
I made PR for that PR: https://github.com/TraceEntertains/Inkay/pull/1

File: src/Notification.cpp [hunk 1]
```diff
@@ -15,41 +15,16 @@
...
-#include <coreinit/thread.h>
-#include <coreinit/title.h>
 #include <notifications/notification_defines.h>
 #include <notifications/notifications.h>
-#include <thread>
 
-static std::unique_ptr<std::thread> sShowHintThread;
-static bool sShutdownHintThread = false;
+void ShowNotification(std::string_view notification) {
+    auto err1 = NotificationModule_SetDefaultValue(NOTIFICATION_MODULE_NOTIFICATION_TYPE_ERROR,
```
Comment (Maschell) [inline src/Notification.cpp:22]: Should be `NOTIFICATION_MODULE_NOTIFICATION_TYPE_INFO`, my bad

**Review by Maschell — CHANGES_REQUESTED** [review]

File: src/Notification.cpp (hunk 1 above)
Comment (TraceEntertains) [reply src/Notification.cpp]: Fixed it

**Comment by TraceEntertains** [comment]
[Inkay-pretendo.zip](https://github.com/PretendoNetwork/Inkay/files/15253804/Inkay-pretendo.zip)
Here's the plugin in a zip file (thanks github...) if anyone wants to test it

**Comment by FourHawk** [comment]
Testing it and everything is working as expected, the message now shows up after boot and patching the wara wara plaza works now

**Comment by Ch3ck3rM0n** [comment]
Using Aroma v19 with your release TraceEntertains and after a few reboots Mario Kart worked but for example with Splatoon I still get Error 101-0502. It seems not to be 100% compatible to me for the moment.

**Comment by FourHawk** [comment]
Error code 101-0502 Normally just means that the server is overloaded, idk about 101-2210 however

**Comment by ItzSwirlz** [comment]
It seems to be working for me on Aroma Beta 18, yesterday night I was able to play a Splatoon match.

**Comment by TraceEntertains** [comment]
@jonbarrow PR generally seems to be working on Aroma Beta 17+, and its been sitting for a few days now, is it possible you could review this when you get the chance?

File: src/Notification.cpp [hunk 2]
```diff
@@ -15,41 +15,17 @@
...
-#include <thread>
 
-static std::unique_ptr<std::thread> sShowHintThread;
-static bool sShutdownHintThread = false;
+void ShowNotification(std::string_view notification) {
+    auto err1 = NotificationModule_SetDefaultValue(NOTIFICATION_MODULE_NOTIFICATION_TYPE_INFO,
+                                                   NOTIFICATION_MODULE_DEFAULT_OPTION_KEEP_UNTIL_SHOWN, true);
+    auto err2 = NotificationModule_SetDefaultValue(NOTIFICATION_MODULE_NOTIFICATION_TYPE_INFO,
+                                                   NOTIFICATION_MODULE_DEFAULT_OPTION_DURATION_BEFORE_FADE_OUT,
+                                                   15.0f);
```
Comment (ashquarky) [inline src/Notification.cpp:26]: Does "SetDefaultValue" affect other plugins too? I've had a look at NotificationModule code and I'm not sure. https://github.com/wiiu-env/libnotifications/blob/75ce22a0f6f670f1fa9a55fb0780f650e6df6701/source/utils.cpp#L320
cc @Maschell

File: src/config.cpp [hunk 3]
```diff
@@ -246,3 +227,38 @@ WUPS_CONFIG_CLOSED() {
...
+	}
+	else if (storageRes != WUPS_STORAGE_ERROR_SUCCESS) {
+		DEBUG_FUNCTION_LINE("Failed to get bool %s (%d)", WUPSStorageAPI_GetStatusStr(storageRes), storageRes);
+	}
+
+	// Save storage
+    if (WUPSStorageAPI::SaveStorage() != WUPS_STORAGE_ERROR_SUCCESS) {
+        DEBUG_FUNCTION_LINE("Failed to save storage");
+    }
+}
```
Comment (ashquarky) [inline src/config.cpp:264]: I'm going to assume this is fine. As long as the previous config files are compatible we should be gucci.
(i.e. someone connected to Nintendo on the old plugin continues being connected to Nintendo after updating)

File: src/main.cpp [hunk 4]
```diff
@@ -188,27 +194,28 @@ INITIALIZE_PLUGIN() {
...
     remove_matchmaking_patches();
 
-    WHBLogUdpDeinit();
     Mocha_DeInitLibrary();
     NotificationModule_DeInitLibrary();
     FunctionPatcher_DeInitLibrary();
+	
+	WHBLogModuleDeinit();
+	WHBLogCafeDeinit();
+    WHBLogUdpDeinit();
```
Comment (ashquarky) [inline src/main.cpp:207]: My understanding is that you're meant to use LogModule *or* the Cafe/UDP ones?
Also spacing but I'll fix that on merge

**Review by ashquarky — COMMENTED** [review]
Seems fine though I have a few question marks. Happy to merge after this though

File: src/Notification.cpp (hunk 2 above)
Comment (Maschell) [reply src/Notification.cpp:26]: It does not

File: src/main.cpp (hunk 4 above)
Comment (TraceEntertains) [reply src/main.cpp:207]: I'm honestly not sure tbh? I kinda just went by the example plugin for beta 17+ and don't really know what it's for, but it didn't seem to affect anything, so I'll remove it in the next commit

Comment (Maschell) [reply src/main.cpp:207]: > My understanding is that you're meant to use LogModule _or_ the Cafe/UDP ones? Also spacing but I'll fix that on merge

In my plugins I first try go init the logging module and if that fails it'll fallback to udp/cafe logging

File: src/config.cpp (hunk 3 above)
Comment (TraceEntertains) [reply src/config.cpp:264]: It should be fine, config save file formats are the same except the config int is now written as a bool (doesn't matter though as it treats ints and bools the same)

**Review by ItzSwirlz — APPROVED** [review]
Just tested with the latest commit and appears to be working fine for me - notification is showing, got 101-0502 (server overload which given Pretendo is a good sign the plugin works)

**Comment by ashquarky** [comment]
Alright we'll call it good

**Pull request merged**